cfpb/cfgov-refresh
# Add feedback variants that record referring page

Owning a Home needs two feedback variants that can reside on their own pages. 
- One is a feedback page that can be referred to from other pages that captures the referrer and allows free-form feedback.
- The second captures referrer, solicits owning-a-home feedback and asks for more user information, including email.

## Additions

- Add two forms to handle variants.

## Changes
- Adapt the Feedback model, form handler and template to the variations.

## Testing

- Fire up locally, run `migrate --fake` 
- Create a feedback page under Owning a Home using the slug 'feedback' and delete the default type for 'was it helpful' and 'radio intro'
- check out http://localhost:8000/owning-a-home/feedback/
- Try entering feedback with no comment
- Try entering feedback with a comment. The success redirect will 404 because page.url points to www, but if you arrow back to the previous page you should see a success message.

## Review

- @cfpb/cfgov-backends @cfpb/cfgov-frontends 


## Todos
- This reduces repeated model, handler and template files but makes the admin more complicated because of unused fields in certain cases.
- Adding more tests but wanted to get eyes on this approach.
- OAH team is deciding on content to add to the sidebar at /owning-a-home/feedback/
